lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -689,8 +696,18 @@ def _on_set_password_action(self, event: ActionEvent):
             self._put_or_update_internal_user_leader(user_name, password)
             label = self.secrets.password_key(user_name)
             event.set_results({label: password})
+            # We know we are already running for MAIN_ORCH. and its leader unit
+            self.peer_cluster_provider.refresh_relation_data(event)
         except OpenSearchError as e:
             event.fail(f"Failed changing the password: {e}")
+        except RuntimeError as e:
```

Comment:
I understand why you're doing it, but it's not very pretty... we can keep it until we find a better way - or maybe we get rid of defer in the refresh_relation_data at some point

Reply:
Agreed, I have an `if` down to check if we are dealing with the defer-related Runtime error or not, but still, it depends on the message internals. It is not nice but it is the only way for now